### ISSUE TITLE
ZipParameters.clone() does not work

### ISSUE THREAD

**Issue by jeffrey-easyesi** [srikanth-lingala/zip4j#421]
ZipParameters has a public `clone()` method, but it doesn't work as (since commit 4c50a6a5) the class does not implement `Cloneable`.

```
  Exception java.lang.CloneNotSupportedException: net.lingala.zip4j.model.ZipParameters
        at Object.clone (Native Method)
        at ZipParameters.clone (ZipParameters.java:208)
```

**Comment by srikanth-lingala** [comment]
`ZipParameters` does not implement `Cloneable` anymore because `Cloneable` is generally considered to be broken and not implemented correctly in Java. Instead `ZipParameters` now has constructor that does exactly that - it creates a new instance of `ZipParameters` from the input `zipParameters`. Just call `new ZipParameters(zipParameters)` where `zipParameters` is your existing instance that you want to clone.

References:
https://blog.jcranky.com/2009/08/26/why-not-use-java-clone-method/
https://howtodoinjava.com/java/cloning/cloneable-interface-is-broken-in-java/

**Comment by jeffrey-easyesi** [comment]
OK; the public `clone()` override should be removed then as it doesn't work and is just misleading.

**Comment by srikanth-lingala** [comment]
You are right. I will take it out.

**Comment by srikanth-lingala** [comment]
Fixed in v2.11.0 released today
